pgrx-tests/src/framework.rs

```diff
@@ -467,6 +530,11 @@ fn start_pg(loglines: LogLines) -> eyre::Result<String> {
             }
         }
 
+        cmd.arg(postmaster_path);
+        cmd
+    } else if let Some(runas) = get_runas() {
+        let mut cmd = Command::new("sudo");
+        cmd.arg("-u").arg(runas);
```

Comment:
Maybe, instead of using `sudo` here, we could execute `id -u "$runas"`, and then, having obtained the relevant UID, use [`<Command as std::os::unix::process::CommandExt>::uid`](https://doc.rust-lang.org/std/os/unix/process/trait.CommandExt.html#required-methods)? Then we wouldn't need to `sudo` just to kill our child, later.

If this doesn't work, then the current pattern will do for now.

Reply:
> Sets the child process’s user ID. This translates to a setuid call in the child process. Failure in the setuid call will cause the spawn to fail.

You'd have to already be root in order for that to work.  Or `cargo-pgrx.exe` would need to be setuid, and that's a bad idea.

Reply:
That's fair. Then yes, let's not use a setuid binary.